x-pack/plugins/snapshot_restore/public/app/services/http/use_request.ts

```diff
@@ -43,10 +43,10 @@ interface UseRequest extends SendRequest {
   interval?: number;
 }
 
-export const useRequest = ({ path, method, body, interval }: UseRequest) => {
+export const useRequest = ({ path, method, body, interval, initialData }: UseRequest) => {
   const [error, setError] = useState<null | any>(null);
-  const [loading, setLoading] = useState<boolean>(false);
-  const [data, setData] = useState<any>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [data, setData] = useState<any>(initialData);
```

Comment:
@jen-huang The hardcoded `[]` value caused errors in situations where I either expected this value to be `undefined` or an object initially, so I added the ability to specify it per request.

Reply:
good catch